From 74b6a0b7ed9c55f82f91c9023fc8be972659125f Mon Sep 17 00:00:00 2001
From: Pawel Zalewski <pzalewski@thegoodpenguin.co.uk>
Date: Wed, 28 Aug 2024 17:02:50 +0100
Subject: [PATCH 2/4] check

Signed-off-by: Pawel Zalewski <pzalewski@thegoodpenguin.co.uk>
---
 src/libs/zbxembed/httprequest.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/libs/zbxembed/httprequest.c b/src/libs/zbxembed/httprequest.c
index 05cb6a7..3eb13a3 100644
--- a/src/libs/zbxembed/httprequest.c
+++ b/src/libs/zbxembed/httprequest.c
@@ -128,6 +128,8 @@ static duk_ret_t	es_httprequest_dtor(duk_context *ctx)
 
 		env->http_req_objects--;
 
+    zabbix_log(LOG_LEVEL_WARNING, "%s: DESTRUCTING HTTP REQUEST @ %p", __func__, request);
+
 		if (NULL != request->headers)
 			curl_slist_free_all(request->headers);
 		if (NULL != request->handle)
@@ -380,6 +382,7 @@ static duk_ret_t	es_httprequest_query(duk_context *ctx, const char *http_request
 	{
 		err_index = duk_push_error_object(ctx, DUK_RET_EVAL_ERROR, "cannot get URL: %s.",
 				curl_easy_strerror(err));
+    //curl_easy_cleanup(request->handle);
 		goto out;
 	}
 out:
@@ -389,7 +392,7 @@ out:
 	if (-1 != err_index)
 		return duk_throw(ctx);
 
-	duk_push_string(ctx, request->data);
+	duk_push_lstring(ctx, request->data, request->data_offset);
 
 	return 1;
 }
-- 
2.45.1

